src/OpenTelemetry.Contrib.Extensions.AWSXRay/Resources/Http/ServerCertificateValidationProvider.cs

```diff
@@ -0,0 +1,156 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.IO;
+using System.Net.Security;
+using System.Security.Cryptography.X509Certificates;
+
+namespace OpenTelemetry.Contrib.Extensions.AWSXRay.Resources.Http
+{
+    internal class ServerCertificateValidationProvider
+    {
+        private static readonly ServerCertificateValidationProvider InvalidProvider =
+            new ServerCertificateValidationProvider(null);
+
+        private readonly X509Certificate2Collection trustedCertificates;
+
+        private ServerCertificateValidationProvider(X509Certificate2Collection trustedCertificates)
+        {
+            if (trustedCertificates == null)
+            {
+                this.trustedCertificates = null;
+                this.ValidationCallback = null;
+                return;
+            }
+
+            this.trustedCertificates = trustedCertificates;
+            this.ValidationCallback = (sender, cert, chain, errors) =>
+                this.ValidateCertificate(new X509Certificate2(cert), chain, errors);
+        }
+
+        public RemoteCertificateValidationCallback ValidationCallback { get; }
+
+        public static ServerCertificateValidationProvider FromCertificateFile(string certificateFile)
+        {
+            if (!File.Exists(certificateFile))
+            {
+                AWSXRayEventSource.Log.FailedToValidateCertificate(nameof(ServerCertificateValidationProvider), "Certificate File does not exist");
+                return InvalidProvider;
+            }
+
+            var trustedCertificates = new X509Certificate2Collection();
+            if (!LoadCertificateToTrustedCollection(trustedCertificates, certificateFile))
+            {
+                AWSXRayEventSource.Log.FailedToValidateCertificate(nameof(ServerCertificateValidationProvider), "Failed to load certificate in trusted collection");
+                return InvalidProvider;
+            }
+
+            return new ServerCertificateValidationProvider(trustedCertificates);
+        }
+
+        private static bool LoadCertificateToTrustedCollection(X509Certificate2Collection collection, string certFileName)
+        {
+            try
+            {
+                collection.Import(certFileName, null, (X509KeyStorageFlags)32);
+                return true;
+            }
+            catch (Exception)
+            {
+                return false;
+            }
+        }
+
+        private bool ValidateCertificate(X509Certificate2 cert, X509Chain chain, SslPolicyErrors errors)
+        {
```

Comment:
@srprash What's your thoughts on this?

Reply:
Yeah, it may be a bit overkill, but since this validation will be done only 1 time on `TraceProvider` initialization, it should be okay. BTW, do you by any chance know what's the performance impact of this validation?

Reply:
We have not done performance measurement, but it should be neglectable as we are not doing much more than standard certificate validation will do - we try to build second chain (one chain is already built by normal check), validate results and scan certificate keys to match certificate on disk. I'd expect it may be up to 3 time slower then normal SSL, but it never will be on hot path - and in our case, it will be only at init time.

Reply:
Sounds good. Thanks!